### PR TITLE
ENH: update, rework, and document env template

### LINKED ISSUE #251: Update template with newest practices
## Expected Behavior

Template should be up to date so when I create rixpython it's already fully correct

## Current Behavior

It's out of date

## Possible Solution

Compare xpp, xcs, cxi, mfx, mec environment files and pick the best attributes for the template

### PR DESCRIPTION
**Pull request by ZLLentz**
## Description

- Update template's version to latest
- Include local hard drive loading
- Add various dev comments

## Motivation and Context

- Might as well clean up here before making the next new one
- closes #251 
- works toward #252 

## How Has This Been Tested?

Runs as expected after creating a new tstpython in a test area on psbuild-rhel7-01

### CODE REVIEW

File: hutch_python/cookiecutter/{{ cookiecutter.hutch }}/{{ cookiecutter.hutch }}env [hunk 1]
```diff
@@ -1,17 +1,44 @@
...
 
+# Clear the user's old env variables
 unset PYTHONPATH
 unset LD_LIBRARY_PATH
 
-source "${CONDA_BASE}/etc/profile.d/conda.sh"
-conda activate "${CONDA_ENVNAME}"
+# Helpers
+LOCAL_ENV="${LOCAL_BASE}/${CONDA_ENVNAME}"
 HERE=`dirname $(readlink -f $0)`
```
Comment (ZryletTC) [inline hutch_python/cookiecutter/{{ cookiecutter.hutch }}/{{ cookiecutter.hutch }}env:22]:
```suggestion
HERE=`dirname $(readlink -f $BASH_SOURCE)`
```

File: hutch_python/cookiecutter/{{ cookiecutter.hutch }}/{{ cookiecutter.hutch }}env [hunk 2]
```diff
@@ -1,17 +1,44 @@
...
+
+# Standard dev path: local only
 export PYTHONPATH="${HERE}:${HERE}/dev/devpath"
+# Expanded hutch-python dev common path: changes specifically for hutch-python
+#export PYTHONPATH="${HERE}:${HERE}/dev/devpath:${HERE}/../common/dev/devpath"
+# Including dev_conda path: dev changes for other apps
+#export PYTHONPATH="${HERE}:${HERE}/dev/devpath:/cds/group/pcds/pyps/apps/dev/pythonpath/"
+
+# Activates pydaq, pycdb, pyami
 source pcdsdaq_lib_setup
```
Comment (ZryletTC) [inline hutch_python/cookiecutter/{{ cookiecutter.hutch }}/{{ cookiecutter.hutch }}env:42]: I've been having issues with this line manually sourcing env files with local installs. Can you give it a try please?

File: hutch_python/cookiecutter/{{ cookiecutter.hutch }}/{{ cookiecutter.hutch }}env [hunk 3]
```diff
@@ -1,17 +1,44 @@
...
-export CONDA_BASE="/reg/g/pcds/pyps/conda/py36"
+# pcds-envs version
+export CONDA_ENVNAME="pcds-3.4.2"
+# In case you want dev dev
+#export CONDA_ENVNAME="dev"
+
+# Settings
 export HUTCH="{{ cookiecutter.hutch }}"
+export LOCAL_BASE="/u1/{{ cookiecutter.hutch }}opr/conda_envs"
+export NFS_BASE="/cds/group/pcds/pyps/conda/py36"
```
Comment (ZryletTC) [inline hutch_python/cookiecutter/{{ cookiecutter.hutch }}/{{ cookiecutter.hutch }}env:13]:
```suggestion
export WEKA_BASE="/cds/group/pcds/pyps/conda/py36"
```
also below

**Review by ZryletTC — APPROVED** [review]
👍

File: hutch_python/cookiecutter/{{ cookiecutter.hutch }}/{{ cookiecutter.hutch }}env (hunk 2 above)
Comment (ZLLentz) [reply hutch_python/cookiecutter/{{ cookiecutter.hutch }}/{{ cookiecutter.hutch }}env:42]: Can you expand a bit? I'm not sure exactly what you mean.

File: hutch_python/cookiecutter/{{ cookiecutter.hutch }}/{{ cookiecutter.hutch }}env (hunk 3 above)
Comment (ZLLentz) [reply hutch_python/cookiecutter/{{ cookiecutter.hutch }}/{{ cookiecutter.hutch }}env:13]: Good idea

File: hutch_python/cookiecutter/{{ cookiecutter.hutch }}/{{ cookiecutter.hutch }}env (hunk 2 above)
Comment (ZryletTC) [reply hutch_python/cookiecutter/{{ cookiecutter.hutch }}/{{ cookiecutter.hutch }}env:42]: Sorry yeah that was kinda vague. Here's an example:
```
xpp-control:xpp pennebak$ source xppenv
Loading local disk python env pcds-3.4.2
bash: pcdsdaq_lib_setup: No such file or directory
(pcds-3.4.2) xpp-control:xpp pennebak$ ssh ioc-xpp-usr01
```

Comment (ZLLentz) [reply hutch_python/cookiecutter/{{ cookiecutter.hutch }}/{{ cookiecutter.hutch }}env:42]: thanks I can look into that

Comment (ZLLentz) [reply hutch_python/cookiecutter/{{ cookiecutter.hutch }}/{{ cookiecutter.hutch }}env:42]: This doesn't happen on my session at all- it just proceeds normally with no apparent error.

Comment (ZryletTC) [reply hutch_python/cookiecutter/{{ cookiecutter.hutch }}/{{ cookiecutter.hutch }}env:42]: Ok, something must just be wrong with my setup

**Review by klauer — APPROVED** [review]
Looking good 👍 Merge at will if ready.

**Pull request merged**